docs/DeveloperGuide.md

```diff
@@ -27,12 +167,71 @@
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+- Runs on any OS with Java 17+.
+- Handles up to 1000 entries without noticeable lag.
+- Should respond to commands within 1 second.
+- CLI-based, with no need for internet connection.
 
 ## Glossary
 
-* *glossary item* - Definition
+| Term | Definition |
+|------|------------|
+| CLI | Command-Line Interface |
+| Command Pattern | A design pattern in which each action is encapsulated in its own class |
+| Category | A label like Food, Transport, Job, etc., used to group entries |
+
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+The following steps help a tester verify the correctness of features:
```

Comment:
It would be nice if for the following section - you can specify the arguments in "<>" angle brackets you expect the user to input instead of giving some dummy data. The dummy data can be given as an example for better understanding. For example:
1. Add an expense
 Expected input - add <item> <cost> <type of item> <date/time>
 Example - add Coffee 4.5 Drink 2025-04-01